src/main/java/net/snowflake/ingest/connection/ServiceResponseHandler.java

```diff
@@ -128,49 +106,98 @@ public static HistoryRangeResponse unmarshallHistoryRangeResponse(HttpResponse r
       throw new IllegalArgumentException();
     }
 
-    // Grab the status line
-    StatusLine line = response.getStatusLine();
-    if (!isStatusOK(line)) {
-      // A network issue occurred!
-      LOGGER.warn(
-          "Exceptional Status Code found in " + "unmarshallHistoryRangeResponse - {}",
-          line.getStatusCode());
+    // handle the exceptional status code
+    handleExceptionalStatus(response);
 
-      // handle the exceptional status code
-      handleExceptionalStatus(line, response);
-      return null;
```

Comment:
Was this unreachable here? Would we have sent `IngestResponseException` in all non ok responses?

Reply:
yes, we sent Exception for all non ok responses.